test/WebSites/RazorPageExecutionInstrumentationWebSite/Startup.cs

```diff
@@ -22,24 +20,28 @@ public void ConfigureServices(IServiceCollection services)
 
             // Add MVC services to the services container.
             services.AddMvc();
-
-            // Make instrumentation data available in views.
-            services.AddScoped<TestPageExecutionListenerFeature, TestPageExecutionListenerFeature>();
-            services.AddScoped<IHoldInstrumentationData>(
-                provider => provider.GetRequiredService<TestPageExecutionListenerFeature>().Holder);
         }
 
         public void Configure(IApplicationBuilder app)
         {
+            var listener = new RazorPageDiagnosticListener();
+            var diagnosticSource = app.ApplicationServices.GetRequiredService<DiagnosticListener>();
+            diagnosticSource.SubscribeWithAdapter(listener);
```

Comment:
It does matter because users look at these sites for inspiration. If we can't set up a per-request diagnostic sink (as I was able to do w/ the old approach), something is wrong here. Suggest we find a way to solve this before pushing this PR.